cmd/substreams/init.go

```diff
@@ -172,8 +172,63 @@ func runSubstreamsInitE(cmd *cobra.Command, args []string) error {
 			return fmt.Errorf("failed to call discovery endpoint: %w", err)
 		}
 
-		var options []huh.Option[*pbconvo.DiscoveryResponse_Generator]
+		type BlockchainProtocolSelector struct {
+			Id         string
+			Title      string
+			Generators []*pbconvo.DiscoveryResponse_Generator
+		}
+		titleFromGroup := func(group string) string {
+			switch group {
+			case "evm":
+				return "EVM"
+			case "solana":
+				return "Solana"
+			case "cosmos":
+				return "Cosmos"
+			case "starknet":
+				return "Starknet"
+			}
+
+			return ""
+		}
+
+		var filteredProtocols map[string]*BlockchainProtocolSelector = make(map[string]*BlockchainProtocolSelector)
 		for _, gen := range resp.Msg.Generators {
+			selector, groupExists := filteredProtocols[gen.Group]
+			if groupExists {
+				selector.Generators = append(selector.Generators, gen)
+			} else {
+				filteredProtocols[gen.Group] = &BlockchainProtocolSelector{
+					Id:         gen.Group,
+					Title:      titleFromGroup(gen.Group),
+					Generators: []*pbconvo.DiscoveryResponse_Generator{gen},
+				}
+			}
+		}
+
+		var protocolOptions []huh.Option[*BlockchainProtocolSelector]
```

Comment:
It's good practice to pre-allocate the array when you know it's fixed size, like here.
```suggestion
	protocolOptions := make([]huh.Option[*BlockchainProtocolSelector], 0, len(filteredProtocols))
```